tests/test_sqlPreFlight.py

```diff
@@ -303,6 +303,40 @@ def testPreFlightSkyMapUnits(self):
         rows = list(rows)
         self.assertEqual(len(rows), 0)
 
+    def testPreFlightSpatialMatch(self):
+        """Test involving spacial match using join tables.
+
+        Note that realistic test needs a resonably-defined SkyPix and regions
+        in registry tables which is hard to implement in this simple test.
+        So we do not actually fill registry with any data and all queries will
+        return empty result, but this is still useful for coverage of the code
+        that generates query.
+        """
+        registry = self.registry
+
+        # dataset types
+        registry.makeRun(collection="test")
+        storageClass = StorageClass("testDataset")
+        registry.storageClasses.registerStorageClass(storageClass)
+
+        calexpType = DatasetType(name="CALEXP", dataUnits=("Camera", "Visit", "Sensor"),
+                                 storageClass=storageClass)
```

Comment:
Im not an expert on the new system yet, but why is this being set up as a storage class with type Object, rather than the actual Calexp type?, same for the coadd below. Is it that it just dosn't matter?

Reply:
Perhaps I am not understanding how the new system works under the hood

Reply:
Not sure if I understand question - `storageClass` defines how objects are stored/formatted, and it is associated with DatasetType so all Datasets of the same DatasetType use same storageClass. In a sense there is no actual Calexp type, but there is a "calexp" DatasetType with storageClass something like "ExposureF".

In this particular case the actual names do not matter and they are arbitrary, and storageClass is not even used (we don't do any I/O) but it has to be defined for every DatasetType.